app/src/main/java/org/piramalswasthya/sakhi/ui/home_activity/infant/hbnc/HBNCFormViewModel.kt

```
@@ -0,0 +1,301 @@
+            val apiSchema = repository.getFormSchema(formId)
+            if (apiSchema == null) {
+                Log.e("HBNCFormViewModel", "Schema load failed")
+                return@launch
+            }
+
+            val savedJson = repository.loadFormResponseJson(benId, visitDay)
+            val savedFieldValues: Map<String, Any?> = try {
+                val root = JSONObject(savedJson ?: "")
+                val fieldsJson = root.optJSONObject("fields") ?: JSONObject()
+                fieldsJson.keys().asSequence().associateWith { fieldsJson.opt(it) }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)